app/controllers/admin/manager_invitations_controller.rb

```diff
@@ -1,5 +1,7 @@
 module Admin
   class ManagerInvitationsController < Spree::Admin::BaseController
+    authorize_resource class: false
```

Comment:
This is the first time I see a line like this. What does it mean? I found https://github.com/ryanb/cancan/wiki/authorizing-controller-actions#authorize_resource explaining that it adds a before filter like `authorize!(:create, @manager_invitation || false)`. Why did you add `class: false`?

And why do we need two authorisations? One in this filter and the other one in the create method.

Reply:
`authorize_class: false` seems to be the recommended way to use cancan in controllers that don't have a corresponding model? It tells cancan to look for symbol keys in the permissions list instead of a named class, and I think it skips some of the default behaviour like trying to instantiate a corresponding object on create actions, amongst others.

The additional permissions check inside the create action makes sure the user has permission to make changes to the specific enterprise in question (after we've pulled up that enterprise record) and not just general permission to access the create manager invites functions.

Reply:
https://github.com/ryanb/cancan/wiki/Non-RESTful-Controllers#alternative-authorize_resource